src/Bicep.Core/Emit/ParamsFileWriter.cs

```
@@ -0,0 +1,153 @@
+                    this.EmitExpression(parameter.Value, jsonWriter);
+                    jsonWriter.WriteEndObject();
+                }
+                jsonWriter.WriteEndObject();
+            }
+
+            jsonWriter.WriteEndObject();
+
+            var content = stringWriter.ToString();
+
+            return content.FromJson<JToken>();
+        }
+
+        public void EmitExpression(SyntaxBase syntax, JsonTextWriter jsonWriter)
+        {
+            switch (syntax)
+            {      
+                case BooleanLiteralSyntax booleanLiteralSyntax:
+                {
+                    jsonWriter.WriteValue(booleanLiteralSyntax.Value);
+                    break;
+                }
+                case IntegerLiteralSyntax integerLiteralSyntax:
+                {
+                    jsonWriter.WriteValue(integerLiteralSyntax.Value);
+                    break;
+                }
+                case StringSyntax stringSyntax:
+                {
+                    jsonWriter.WriteValue(string.Join("", stringSyntax.SegmentValues));
+                    break;
+                }
+                case ObjectSyntax objectSyntax:
+                {
+                    jsonWriter.WriteStartObject();
+                    EmitObjectProperties(objectSyntax, jsonWriter);
+                    jsonWriter.WriteEndObject();
+                    break;
+                }
+                case ArraySyntax arraySyntax:
+                {
+                    jsonWriter.WriteStartArray();
+                    foreach (ArrayItemSyntax itemSyntax in arraySyntax.Items)
+                    {
+                        EmitExpression(itemSyntax.Value, jsonWriter);
+                    }
+                    jsonWriter.WriteEndArray();
+                    break;
+                }
+                case NullLiteralSyntax _:
+                {
+                    jsonWriter.WriteNull();
+                    break;
+                }
+            }
```

Better to be explicit and add a `default` switch case to throw an exception - IMO better to know about any missing implementation up front than to silently ignore it.